### ISSUE TITLE
`jx create cluster aws` fails: unable to reach DNS name

### ISSUE THREAD

**Issue by asnowfix** [jenkins-x/jx#825]
I am willing to create a cluster from scratch w/ jx. I have Administrator rights on the AWS account. The cluster seems to be created, but its DNS record never comes up. Here is the execution log:

```
jx create cluster aws --name=kowalskif-3 --nodes=3
Creating cluster...] (3)
running command: kops create cluster --name kowalskif-3.cluster.k8s.local --node-count 3 --authorization RBAC --zones eu-west-3a,eu-west-3b --yes

Kops has created cluster kowalskif-3.cluster.k8s.local it will take a minute or so to startup
You can check on the status in another terminal via the command: kops validate cluster
Waiting for the Cluster configuration...
Loaded Cluster JSON: {"kind":"Cluster","apiVersion":"kops/v1alpha2","metadata":{"name":"kowalskif-3.cluster.k8s.local","creationTimestamp":"2018-05-16T14:05:30Z"},"spec":{"channel":"stable","configBase":"s3://kops-state-store-kowalskif/kowalskif-3.cluster.k8s.local","cloudProvider":"aws","kubernetesVersion":"1.9.3","subnets":[{"name":"eu-west-3a","zone":"eu-west-3a","cidr":"172.20.32.0/19","type":"Public"},{"name":"eu-west-3b","zone":"eu-west-3b","cidr":"172.20.64.0/19","type":"Public"}],"masterPublicName":"api.kowalskif-3.cluster.k8s.local","networkCIDR":"172.20.0.0/16","topology":{"masters":"public","nodes":"public","dns":{"type":"Public"}},"nonMasqueradeCIDR":"100.64.0.0/10","sshAccess":["0.0.0.0/0"],"kubernetesApiAccess":["0.0.0.0/0"],"etcdClusters":[{"name":"main","etcdMembers":[{"name":"a","instanceGroup":"master-eu-west-3a"}]},{"name":"events","etcdMembers":[{"name":"a","instanceGroup":"master-eu-west-3a"}]}],"networking":{"kubenet":{}},"api":{"loadBalancer":{"type":"Public"}},"authorization":{"rbac":{}},"iam":{"legacy":false,"allowContainerRegistry":true}}}
new json: {"apiVersion":"kops/v1alpha2","kind":"Cluster","metadata":{"creationTimestamp":"2018-05-16T14:05:30Z","name":"kowalskif-3.cluster.k8s.local"},"spec":{"api":{"loadBalancer":{"type":"Public"}},"authorization":{"rbac":{}},"channel":"stable","cloudProvider":"aws","configBase":"s3://kops-state-store-kowalskif/kowalskif-3.cluster.k8s.local","docker":{"insecureRegistry":"100.64.0.0/10"},"etcdClusters":[{"etcdMembers":[{"instanceGroup":"master-eu-west-3a","name":"a"}],"name":"main"},{"etcdMembers":[{"instanceGroup":"master-eu-west-3a","name":"a"}],"name":"events"}],"iam":{"allowContainerRegistry":true,"legacy":false},"kubernetesApiAccess":["0.0.0.0/0"],"kubernetesVersion":"1.9.3","masterPublicName":"api.kowalskif-3.cluster.k8s.local","networkCIDR":"172.20.0.0/16","networking":{"kubenet":{}},"nonMasqueradeCIDR":"100.64.0.0/10","sshAccess":["0.0.0.0/0"],"subnets":[{"cidr":"172.20.32.0/19","name":"eu-west-3a","type":"Public","zone":"eu-west-3a"},{"cidr":"172.20.64.0/19","name":"eu-west-3b","type":"Public","zone":"eu-west-3b"}],"topology":{"dns":{"type":"Public"},"masters":"public","nodes":"public"}}}
Updating Cluster configuration to enable insecure docker registries 100.64.0.0/10
Updating the cluster
Rolling update the cluster
Cluster configuration updated
Waiting for the kubernetes cluster to be ready so we can continue...
Unable to connect to the server: EOF
retrying after error: exit status 1
Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
^C
```
 kops, helm & jx downloaded today:

```
$ jx --version
1.2.61

$ kops version
Version 1.9.0 (git-cccd71e67)
```

**Comment by mountaineer123** [comment]
I'm also experiencing this same issue. This has worked for me before, but for the last week, this same issue has happened.

Workaround for me was to wait until cluster is up and ready and then run

jx install --provider aws



```
$ jx --version
1.2.79
$ kops version
Version 1.9.0 (git-cccd71e67)
```

```
$ jx create cluster aws
kops not found
kubectl not found
helm not found
? Missing required dependencies, deselect to avoid auto installing: kops, kubectl, helm
Downloading https://github.com/kubernetes/kops/releases/download/1.9.0/kops-linux-amd64 to /home/ec2-user/.jx/bin/kops.tmp...
Downloaded /home/ec2-user/.jx/bin/kops.tmp
Downloading https://storage.googleapis.com/kubernetes-release/release/v1.10.3/bin/linux/amd64/kubectl to /home/ec2-user/.jx/bin/kubectl.tmp...
Downloaded /home/ec2-user/.jx/bin/kubectl.tmp
Downloading https://storage.googleapis.com/kubernetes-helm/helm-v2.9.1-linux-amd64.tar.gz to /home/ec2-user/.jx/bin/helm.tgz...
? nodes 3
Creating cluster...] (3) 
running command: kops create cluster --name aws1.cluster.k8s.local --node-count 3 --authorization RBAC --zones ap-southeast-1a,ap-southeast-1b,ap-southeast-1c --yes

Kops has created cluster aws1.cluster.k8s.local it will take a minute or so to startup
You can check on the status in another terminal via the command: kops validate cluster
Waiting for the Cluster configuration...
Loaded Cluster JSON: {"kind":"Cluster","apiVersion":"kops/v1alpha2","metadata":{"name":"aws1.cluster.k8s.local","creationTimestamp":"2018-05-24T05:33:51Z"},"spec":{"channel":"stable","configBase":"s3://devopside-kopsstatestore-yfijms51imb1/aws1.cluster.k8s.local","cloudProvider":"aws","kubernetesVersion":"1.9.3","subnets":[{"name":"ap-southeast-1a","zone":"ap-southeast-1a","cidr":"172.20.32.0/19","type":"Public"},{"name":"ap-southeast-1b","zone":"ap-southeast-1b","cidr":"172.20.64.0/19","type":"Public"},{"name":"ap-southeast-1c","zone":"ap-southeast-1c","cidr":"172.20.96.0/19","type":"Public"}],"masterPublicName":"api.aws1.cluster.k8s.local","networkCIDR":"172.20.0.0/16","topology":{"masters":"public","nodes":"public","dns":{"type":"Public"}},"nonMasqueradeCIDR":"100.64.0.0/10","sshAccess":["0.0.0.0/0"],"kubernetesApiAccess":["0.0.0.0/0"],"etcdClusters":[{"name":"main","etcdMembers":[{"name":"a","instanceGroup":"master-ap-southeast-1a"}]},{"name":"events","etcdMembers":[{"name":"a","instanceGroup":"master-ap-southeast-1a"}]}],"networking":{"kubenet":{}},"api":{"loadBalancer":{"type":"Public"}},"authorization":{"rbac":{}},"iam":{"legacy":false,"allowContainerRegistry":true}}}
new json: {"apiVersion":"kops/v1alpha2","kind":"Cluster","metadata":{"creationTimestamp":"2018-05-24T05:33:51Z","name":"aws1.cluster.k8s.local"},"spec":{"api":{"loadBalancer":{"type":"Public"}},"authorization":{"rbac":{}},"channel":"stable","cloudProvider":"aws","configBase":"s3://devopside-kopsstatestore-yfijms51imb1/aws1.cluster.k8s.local","docker":{"insecureRegistry":"100.64.0.0/10"},"etcdClusters":[{"etcdMembers":[{"instanceGroup":"master-ap-southeast-1a","name":"a"}],"name":"main"},{"etcdMembers":[{"instanceGroup":"master-ap-southeast-1a","name":"a"}],"name":"events"}],"iam":{"allowContainerRegistry":true,"legacy":false},"kubernetesApiAccess":["0.0.0.0/0"],"kubernetesVersion":"1.9.3","masterPublicName":"api.aws1.cluster.k8s.local","networkCIDR":"172.20.0.0/16","networking":{"kubenet":{}},"nonMasqueradeCIDR":"100.64.0.0/10","sshAccess":["0.0.0.0/0"],"subnets":[{"cidr":"172.20.32.0/19","name":"ap-southeast-1a","type":"Public","zone":"ap-southeast-1a"},{"cidr":"172.20.64.0/19","name":"ap-southeast-1b","type":"Public","zone":"ap-southeast-1b"},{"cidr":"172.20.96.0/19","name":"ap-southeast-1c","type":"Public","zone":"ap-southeast-1c"}],"topology":{"dns":{"type":"Public"},"masters":"public","nodes":"public"}}}
Updating Cluster configuration to enable insecure docker registries 100.64.0.0/10
Updating the cluster
Rolling update the cluster
Error: Command failed  kops rolling-update cluster --cloudonly --yes
WARNING: Failed to perform rolling upgrade: exit status 1
Cluster configuration updated
Waiting for the kubernetes cluster to be ready so we can continue...
Unable to connect to the server: dial tcp: lookup api-aws1-cluster-k8s-loca-jh2pl9-725864305.ap-southeast-1.elb.amazonaws.com on 172.31.0.2:53: no such host
retrying after error: exit status 1
Unable to connect to the server: dial tcp: lookup api-aws1-cluster-k8s-loca-jh2pl9-725864305.ap-southeast-1.elb.amazonaws.com on 172.31.0.2:53: no such host
.Unable to connect to the server: dial tcp: lookup api-aws1-cluster-k8s-loca-jh2pl9-725864305.ap-southeast-1.elb.amazonaws.com on 172.31.0.2:53: no such host
.Unable to connect to the server: dial tcp: lookup api-aws1-cluster-k8s-loca-jh2pl9-725864305.ap-southeast-1.elb.amazonaws.com on 172.31.0.2:53: no such host
.Unable to connect to the server: dial tcp: lookup api-aws1-cluster-k8s-loca-jh2pl9-725864305.ap-southeast-1.elb.amazonaws.com on 172.31.0.2:53: no such host
.Unable to connect to the server: dial tcp: lookup api-aws1-cluster-k8s-loca-jh2pl9-725864305.ap-southeast-1.elb.amazonaws.com on 172.31.0.2:53: no such host
.Unable to connect to the server: dial tcp: lookup api-aws1-cluster-k8s-loca-jh2pl9-725864305.ap-southeast-1.elb.amazonaws.com on 172.31.0.2:53: no such host
.Unable to connect to the server: dial tcp: lookup api-aws1-cluster-k8s-loca-jh2pl9-725864305.ap-southeast-1.elb.amazonaws.com on 172.31.0.2:53: no such host
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.No resources found.
```

**Comment by asnowfix** [comment]
Going further in the investigation (which is not yet over), it might be a kops-only issue: kops use to fetch the default DNSZoneId for the account & use it at part of the built-in kops cluster template spec. It looks like this is no longer the case: one option is to make the kops template explicit in jx & fetch the necessary fields or report an issue (not sure it is not by design...) to kops. What is the preferred approach @jstrachan ?

**Comment by asnowfix** [comment]
Well, I am pretty sure it is not a kops issue now: running the kops command alone works:

```
kops create cluster --name kowalskif-5.cluster.k8s.local --node-count 3 --authorization RBAC --zones us-east-1a,us-east-1b --yes
```

...and some time after it works:

```
$ kops validate cluster
Using cluster from kubectl context: kowalskif-5.cluster.k8s.local

Validating cluster kowalskif-5.cluster.k8s.local

INSTANCE GROUPS
NAME			ROLE	MACHINETYPE	MIN	MAX	SUBNETS
master-us-east-1a	Master	m3.medium	1	1	us-east-1a
nodes			Node	t2.medium	3	3	us-east-1a,us-east-1b

NODE STATUS
NAME				ROLE	READY
[...]

Your cluster kowalskif-5.cluster.k8s.local is ready
```

That might be a too-short jx timeout waiting for the cluster to come-up.

**Comment by mountaineer123** [comment]
Agree.

Previous versions of jx (just tried 1.2.50) didn't error out if the cluster wasn't ready and could continue with its setup

**Comment by asnowfix** [comment]
Indeed, this is what I am doing: native `kops` followed by `jx install`. I suspect that the timeout that `jx` is using to call `kops validate cluster` might be a bit too short. I'll experiment & let you know the results. That could possibly lead to my first PR to jx to set a new CLI option... ;-)

**Comment by jstrachan** [comment]
it'd be nice to figure out which timeout is timing out; I wondered if it was this one but thats about 5.5 hours already: https://github.com/jenkins-x/jx/blob/master/pkg/jx/cmd/create_cluster_aws.go#L245

maybe its this one? https://github.com/jenkins-x/jx/blob/master/pkg/jx/cmd/create_cluster_aws.go#L237

Having the full output of a timed out `jx create cluster aws` would help narrow down how far its got

**Comment by asnowfix** [comment]
Creation on its way...

```
$ jx create cluster aws --cluster-name kowalskif-6 --zones eu-west-3a,eu-west-3a,eu-west-3c --nodes=3 --rbac=true --domain=george.hpccp.net --default-admin-password=${JENKINS_PASSWORD} --environment-git-owner=gdrs --git-api-token=${GHE_API_TOKEN} --git-provider-url=https://github.azc.ext.hp.com --git-username=${GHE_USER} --username=`git config user.email` --verbose=true
Creating cluster...
running command: kops create cluster --name kowalskif-6.cluster.k8s.local --node-count 3 --authorization RBAC --zones eu-west-3a,eu-west-3a,eu-west-3c --yes
I0524 12:19:28.148080   21272 create_cluster.go:1318] Using SSH public key: /home/kowalskif/.ssh/id_rsa.pub
I0524 12:19:29.661062   21272 create_cluster.go:472] Inferred --cloud=aws from zone "eu-west-3a"
I0524 12:19:30.050736   21272 subnets.go:184] Assigned CIDR 172.20.32.0/19 to subnet eu-west-3a
I0524 12:19:30.050761   21272 subnets.go:184] Assigned CIDR 172.20.64.0/19 to subnet eu-west-3c
I0524 12:19:35.823951   21272 apply_cluster.go:456] Gossip DNS: skipping DNS validation
I0524 12:19:37.548176   21272 executor.go:91] Tasks: 0 done / 79 total; 30 can run
I0524 12:19:41.047535   21272 vfs_castore.go:731] Issuing new certificate: "ca"
I0524 12:19:41.382587   21272 vfs_castore.go:731] Issuing new certificate: "apiserver-aggregator-ca"
I0524 12:19:43.700701   21272 executor.go:91] Tasks: 30 done / 79 total; 25 can run
I0524 12:19:46.599099   21272 vfs_castore.go:731] Issuing new certificate: "kops"
I0524 12:19:46.757246   21272 vfs_castore.go:731] Issuing new certificate: "kubelet-api"
I0524 12:19:47.009299   21272 vfs_castore.go:731] Issuing new certificate: "apiserver-proxy-client"
I0524 12:19:47.153957   21272 vfs_castore.go:731] Issuing new certificate: "kube-proxy"
I0524 12:19:47.799680   21272 vfs_castore.go:731] Issuing new certificate: "kube-scheduler"
I0524 12:19:47.927423   21272 vfs_castore.go:731] Issuing new certificate: "kubelet"
I0524 12:19:48.004012   21272 vfs_castore.go:731] Issuing new certificate: "kubecfg"
I0524 12:19:48.135493   21272 vfs_castore.go:731] Issuing new certificate: "kube-controller-manager"
I0524 12:19:48.448857   21272 vfs_castore.go:731] Issuing new certificate: "apiserver-aggregator"
I0524 12:19:51.437187   21272 executor.go:91] Tasks: 55 done / 79 total; 20 can run
I0524 12:19:53.279438   21272 executor.go:91] Tasks: 75 done / 79 total; 3 can run
I0524 12:19:55.942870   21272 vfs_castore.go:731] Issuing new certificate: "master"
I0524 12:19:57.989560   21272 executor.go:91] Tasks: 78 done / 79 total; 1 can run
I0524 12:19:58.356392   21272 executor.go:91] Tasks: 79 done / 79 total; 0 can run
I0524 12:19:58.831579   21272 update_cluster.go:291] Exporting kubecfg for cluster
kops has set your kubectl context to kowalskif-6.cluster.k8s.local

Cluster is starting.  It should be ready in a few minutes.

Suggestions:
 * validate cluster: kops validate cluster
 * list nodes: kubectl get nodes --show-labels
 * ssh to the master: ssh -i ~/.ssh/id_rsa admin@api.kowalskif-6.cluster.k8s.local
 * the admin user is specific to Debian. If not using Debian please use the appropriate user based on your OS.
 * read about installing addons at: https://github.com/kubernetes/kops/blob/master/docs/addons.md.


Kops has created cluster kowalskif-6.cluster.k8s.local it will take a minute or so to startup
You can check on the status in another terminal via the command: kops validate cluster
Waiting for the Cluster configuration...
Loaded Cluster JSON: {"kind":"Cluster","apiVersion":"kops/v1alpha2","metadata":{"name":"kowalskif-6.cluster.k8s.local","creationTimestamp":"2018-05-24T10:19:30Z"},"spec":{"channel":"stable","configBase":"s3://kops-state-store-kowalskif/kowalskif-6.cluster.k8s.local","cloudProvider":"aws","kubernetesVersion":"1.9.3","subnets":[{"name":"eu-west-3a","zone":"eu-west-3a","cidr":"172.20.32.0/19","type":"Public"},{"name":"eu-west-3c","zone":"eu-west-3c","cidr":"172.20.64.0/19","type":"Public"}],"masterPublicName":"api.kowalskif-6.cluster.k8s.local","networkCIDR":"172.20.0.0/16","topology":{"masters":"public","nodes":"public","dns":{"type":"Public"}},"nonMasqueradeCIDR":"100.64.0.0/10","sshAccess":["0.0.0.0/0"],"kubernetesApiAccess":["0.0.0.0/0"],"etcdClusters":[{"name":"main","etcdMembers":[{"name":"a","instanceGroup":"master-eu-west-3a"}]},{"name":"events","etcdMembers":[{"name":"a","instanceGroup":"master-eu-west-3a"}]}],"networking":{"kubenet":{}},"api":{"loadBalancer":{"type":"Public"}},"authorization":{"rbac":{}},"iam":{"legacy":false,"allowContainerRegistry":true}}}
new json: {"apiVersion":"kops/v1alpha2","kind":"Cluster","metadata":{"creationTimestamp":"2018-05-24T10:19:30Z","name":"kowalskif-6.cluster.k8s.local"},"spec":{"api":{"loadBalancer":{"type":"Public"}},"authorization":{"rbac":{}},"channel":"stable","cloudProvider":"aws","configBase":"s3://kops-state-store-kowalskif/kowalskif-6.cluster.k8s.local","docker":{"insecureRegistry":"100.64.0.0/10"},"etcdClusters":[{"etcdMembers":[{"instanceGroup":"master-eu-west-3a","name":"a"}],"name":"main"},{"etcdMembers":[{"instanceGroup":"master-eu-west-3a","name":"a"}],"name":"events"}],"iam":{"allowContainerRegistry":true,"legacy":false},"kubernetesApiAccess":["0.0.0.0/0"],"kubernetesVersion":"1.9.3","masterPublicName":"api.kowalskif-6.cluster.k8s.local","networkCIDR":"172.20.0.0/16","networking":{"kubenet":{}},"nonMasqueradeCIDR":"100.64.0.0/10","sshAccess":["0.0.0.0/0"],"subnets":[{"cidr":"172.20.32.0/19","name":"eu-west-3a","type":"Public","zone":"eu-west-3a"},{"cidr":"172.20.64.0/19","name":"eu-west-3c","type":"Public","zone":"eu-west-3c"}],"topology":{"dns":{"type":"Public"},"masters":"public","nodes":"public"}}}
Updating Cluster configuration to enable insecure docker registries 100.64.0.0/10
Updating the cluster
Using cluster from kubectl context: kowalskif-6.cluster.k8s.local

I0524 12:20:16.758923   21415 apply_cluster.go:456] Gossip DNS: skipping DNS validation
I0524 12:20:18.430906   21415 executor.go:91] Tasks: 0 done / 79 total; 30 can run
I0524 12:20:19.922112   21415 executor.go:91] Tasks: 30 done / 79 total; 25 can run
I0524 12:20:23.496564   21415 executor.go:91] Tasks: 55 done / 79 total; 20 can run
I0524 12:20:24.802560   21415 executor.go:91] Tasks: 75 done / 79 total; 3 can run
I0524 12:20:25.437311   21415 executor.go:91] Tasks: 78 done / 79 total; 1 can run
I0524 12:20:25.541312   21415 executor.go:91] Tasks: 79 done / 79 total; 0 can run
I0524 12:20:26.004348   21415 update_cluster.go:291] Exporting kubecfg for cluster
kops has set your kubectl context to kowalskif-6.cluster.k8s.local

Cluster changes have been applied to the cloud.


Changes may require instances to restart: kops rolling-update cluster

Rolling update the cluster
Using cluster from kubectl context: kowalskif-6.cluster.k8s.local

NAME			STATUS		NEEDUPDATE	READY	MIN	MAX
master-eu-west-3a	Ready		0		0	1	1
nodes			NeedsUpdate	3		0	3	3
W0524 12:20:31.254111   21479 instancegroups.go:152] Not draining cluster nodes as 'cloudonly' flag is set.
I0524 12:20:31.254129   21479 instancegroups.go:275] Stopping instance "i-009a794023a95aa33", in group "nodes.kowalskif-6.cluster.k8s.local".


W0524 12:24:31.435960   21479 instancegroups.go:184] Not validating cluster as cloudonly flag is set.
W0524 12:24:31.436000   21479 instancegroups.go:152] Not draining cluster nodes as 'cloudonly' flag is set.
I0524 12:24:31.436025   21479 instancegroups.go:275] Stopping instance "i-07e41d8cd3ee72ae3", in group "nodes.kowalskif-6.cluster.k8s.local".
...
```

...not yet completed/failed.

**Comment by jstrachan** [comment]
ah thats looking good - its created the cluster, waited for it to start then modified the JSON (to enable insecure docker registries on service IPs inside the cluster) then its looking like its waiting for ingress to come up

**Comment by asnowfix** [comment]
...then:

```
[...]
       


W0524 12:28:32.946883   21479 instancegroups.go:184] Not validating cluster as cloudonly flag is set.
W0524 12:28:32.947010   21479 instancegroups.go:152] Not draining cluster nodes as 'cloudonly' flag is set.
I0524 12:28:32.947036   21479 instancegroups.go:275] Stopping instance "i-09a731bfc5f2fdaa8", in group "nodes.kowalskif-6.cluster.k8s.local".
W0524 12:32:33.325140   21479 instancegroups.go:184] Not validating cluster as cloudonly flag is set.
I0524 12:32:33.325556   21479 rollingupdate.go:193] Rolling update completed for cluster "kowalskif-6.cluster.k8s.local"!
Cluster configuration updated
Waiting for the kubernetes cluster to be ready so we can continue...
Unable to connect to the server: EOF
retrying after error: exit status 1
Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
.Unable to connect to the server: EOF
^C
`

**Comment by mountaineer123** [comment]
Seems commit [8b97666](https://github.com/jenkins-x/jx/commit/8b97666778ab2e4da337b45e3bae8eed189e246d) for release 1.2.58 added error handling if kops validate cluster fails, and doesn't allow jx install to continue.

**Comment by rawlingsj** [comment]
@mountaineer123 are you thinking we need to log a warning and try to continue with the installation?

**Comment by mountaineer123** [comment]
@rawlingsj Suggest not to error out but to wait until validation succeeds within a greater grace period. Noticed in my scenario, validation eventually succeeds after a few minutes.

**Comment by Salaboy** [comment]
@asnowfix weird.. myself and another team member managed to install jx in AWS using Kops with not too much trouble.. The only thing that happened consistently to both of us is that after a while we couldn't connect to it due to a DNS resolution problem.
Do you really need the route53 configuration? 

Can you share the value of the Availability Zones env variable? that was one of the problems that I found.

**Comment by asnowfix** [comment]
@Salaboy for the test above, I was using the CLI option `--zones=eu-west-3a,eu-west-3b,eu-west-3c` and I also had `AWS_REGION=eu-west-3`. Route53 being AWS's DNS configuration service, I am not sure how I could get DNS working with the requested domain without Route53.

**Comment by asnowfix** [comment]
FWIW, I made a `diff -uw` between a `kops get cluster` between a cluster that we create ourselves (on which we can do `jx install` & which result in jenkins URL to be reachable on `.nip.io` domain, renamed `ok.mydomain.com` in the diff below) and a `jx`-created cluster which never comes up (the kowalskif-9 above, renamed `nok.mydomain.com` in the diff below).

Few highlights of the below verbatim diff:

- the non working cluster has a `dns: {}` entry while the working one has a specific zone: `dnsZone: Z39U3D7PTJO5PL` 
- The working cluster has `loadBalancer:\n  type: Public` and `clusterDNSDomain: cluster.local`
- The non-working cluster uses `networking: kubenet: {}` , while the working one uses `networking: weave: {}`

Does the above rings any bell? Any working cluster configuration (`kops get cluster foo.bar --output=yaml`) you cloud share? 

Here is the verbatim diff

```diff
--- ./nok.mydomain.com.yaml	2018-05-31 12:21:45.083001000 +0200
+++ ok.mydomain.com.config.yaml	2018-05-25 15:51:13.468420500 +0200
@@ -1,56 +1,86 @@
 apiVersion: kops/v1alpha2
 kind: Cluster
 metadata:
-  creationTimestamp: 2018-05-31T09:58:36Z
-  name: nok.mydomain.com
+  creationTimestamp: 2018-05-16T17:01:03Z
+  name: ok.mydomain.com
 spec:
   api:
-    dns: {}
+    loadBalancer:
+      type: Public
   authorization:
     rbac: {}
   channel: stable
   cloudProvider: aws
-  configBase: s3://nok-mybucket/nok.mydomain.com
+  clusterDNSDomain: cluster.local
+  configBase: s3://ok-mybucket/ok.mydomain.com
+  configStore: s3://ok-mybucket/ok.mydomain.com
+  dnsZone: Z39U3D7PTJO5PL
   docker:
     insecureRegistry: 100.64.0.0/10
     logDriver: ""
   etcdClusters:
   - etcdMembers:
-    - instanceGroup: master-eu-west-3a
+    - encryptedVolume: true
+      instanceGroup: master-us-east-1a
       name: a
     name: main
+    version: 3.1.10
   - etcdMembers:
-    - instanceGroup: master-eu-west-3a
+    - encryptedVolume: true
+      instanceGroup: master-us-east-1a
       name: a
     name: events
+    version: 3.1.10
   iam:
     allowContainerRegistry: true
     legacy: false
+  keyStore: s3://ok-mybucket/ok.mydomain.com/pki
+  kubeAPIServer:
+    admissionControl:
+    - NamespaceLifecycle
+    - LimitRanger
+    - ServiceAccount
+    - PersistentVolumeLabel
+    - DefaultStorageClass
+    - DefaultTolerationSeconds
+    - MutatingAdmissionWebhook
+    - ValidatingAdmissionWebhook
+    - ResourceQuota
+    - Initializers
+    - NodeRestriction
+    - Priority
+    auditLogMaxAge: 10
+    auditLogMaxBackups: 1
+    auditLogMaxSize: 100
+    auditLogPath: /var/log/kube-apiserver-audit.log
+    runtimeConfig:
+      admissionregistration.k8s.io/v1beta1: "true"
   kubernetesApiAccess:
   - 0.0.0.0/0
-  kubernetesVersion: 1.9.3
-  masterPublicName: api.nok.mydomain.com
+  kubernetesVersion: 1.9.6
+  masterInternalName: api.internal.ok.mydomain.com
+  masterPublicName: api.ok.mydomain.com
   networkCIDR: 172.20.0.0/16
   networking:
-    kubenet: {}
+    weave: {}
   nonMasqueradeCIDR: 100.64.0.0/10
+  secretStore: s3://ok-mybucket/ok.mydomain.com/secrets
+  serviceClusterIPRange: 100.64.0.0/13
   sshAccess:
-  - 0.0.0.0/0
+  - 15.0.0.0/9
   subnets:
   - cidr: 172.20.32.0/19
-    name: eu-west-3a
-    type: Public
-    zone: eu-west-3a
-  - cidr: 172.20.64.0/19
-    name: eu-west-3b
-    type: Public
-    zone: eu-west-3b
-  - cidr: 172.20.96.0/19
-    name: eu-west-3c
-    type: Public
-    zone: eu-west-3c
+    name: private-us-east-1a
+    type: Private
+    zone: us-east-1a
+  - cidr: 172.20.0.0/22
+    name: utility-us-east-1a
+    type: Utility
+    zone: us-east-1a
   topology:
+    bastion:
+      bastionPublicName: bastion.ok.mydomain.com
     dns:
       type: Public
-    masters: public
-    nodes: public
+    masters: private
+    nodes: private
```

**Comment by asnowfix** [comment]
BTW, my attempt to use the combination `--name <SHORTNAME> --domain <FQDN>` came from a CLI example by @jstrachan  in https://github.com/jenkins-x/jx/issues/681. Looks like it is working, but DNS never comes up.

**Comment by ryandawsonuk** [comment]
> Using --name=kowalskif-8 --domain=kowalskif-8.george.hpccp.net seems to work (goes to the prompt to get the Jenkins API token), but nothing is provisioned in Route53 for that domain (I checked in the AWS console), so the URL is not reachable

I hit a similar problem with 'jx install --provider=aws' on existing cluster (even one using dns created with kops and  --dns-zone, which I think is the same as what you were doing with --domain on jx create cluster). From there I choose yes to installing an ingress controller, no to resolving an IP address and then I enter my domain (which is also the suffix of my cluster name) and everything looks great until I am told to go to http://jenkins.jx.<DOMAIN>/me/configure ... To get that working I had to go to EC2 in AWS console and find the ELB that was created (EC2 -> Load Balancing -> Load Balancers - found it by timestamp) then get its DNS name record, go to Route53, choose the hosted zone for the domain and create an alias-type Record Set for *.domain (the wildcard dns) and the ELB dns name.

So basically I think I'm agreeing that the route53 isn't created and you would expect jx to create it, especially in the case where you've chosen --domain.

Seems a closely-related issue is https://github.com/jenkins-x/jx/issues/673 as that's also about ELB being created with the necessary route53 Record Set

**Comment by asnowfix** [comment]
Thanks for the tips @ryandawsonuk!

Asking for JX itself (not with Kops) to create the ELB is probably over-asking: the clusters we create here are via Kops only, using a cluster spec (you can see the spec diff above), so when JX is given  this —domain option, it should probably use à Kops cluster spec that would be creating the ELB & hook it to Route53. Thoughts?

**Comment by ryandawsonuk** [comment]
You mean jx should have a spec like that built into it or the user should supply one or both?

**Comment by asnowfix** [comment]
@ryandawsonuk I mean that `jx` should probably have a built-in spec that allows for this ELB creation + Route53 setup when a `--domain` is provided. Of course, letting the user provide his own spec could be an option too, but I am wondering if that would not drift too far away from the `jx` objective of simplifying things as much as possible.

**Comment by asnowfix** [comment]
...on the other hand, maybe adding a user-provided spec will help defining what could be that built-in jx spec. Just thinking out loud here...

**Comment by ccojocar** [comment]
@asnowfix Is this still an issue with the current version of Jenkins X?

**Comment by ccojocar** [comment]
@asnowfix I am going to close this issue. Feel free to open another issue if you are still facing this problem. Thanks
